test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -24,49 +25,328 @@ void SignalPhase(const std::wstring& phaseEventName)
     }
 }
 
-std::wstring BuildNotificationPayload(std::wstring channel)
+void initUnitTestMapping()
 {
-    std::wstring channelUri = L"\"ChannelUri\":\"" + channel + L"\"";
-    std::wstring x_wns_type = L"\"X_WNS_Type\": \"wns/raw\"";
-    std::wstring contentType = L"\"Content_Type\": \"application/octet-stream\"";
-    std::wstring payload = L"\"Payload\": \"<toast></toast>\"";
-    std::wstring delay = L"\"Delay\": \"false\"";
-    return { L"{" + channelUri + L"," + x_wns_type + L"," + contentType + L"," + payload + L"," + delay + L"}" };
+    switchMapping["ChannelRequestUsingNullRemoteId"] = UnitTest::channelRequestUsingNullRemoteId;
+    switchMapping["ChannelRequestUsingRemoteId"] = UnitTest::channelRequestUsingRemoteId;
+    switchMapping["MultipleChannelRequestUsingSameRemoteId"] = UnitTest::multipleChannelRequestUsingSameRemoteId;
+    switchMapping["MultipleChannelRequestUsingMultipleRemoteId"] = UnitTest::multipleChannelRequestUsingMultipleRemoteId;
+    switchMapping["ThreeChannelRequestUsingSameRemoteId"] = UnitTest::threeChannelRequestUsingSameRemoteId;
+}
+
+bool ChannelRequestUsingNullRemoteId()
+{
+    winrt::hresult hr = S_OK;
+
+    try
+    {
+        auto channelOperation = PushNotificationManager::CreateChannelAsync(GUID_NULL).get();
+    }
+    catch (...)
+    {
+        auto channelRequestException = hresult_error(to_hresult());
+        hr = channelRequestException.code();
+    }
+
+    if (hr == E_INVALIDARG)
+        return true;
+    else
+        return false;
 }
 
-void sendRequestToServer(winrt::hstring channel)
+bool ChannelRequestUsingRemoteId()
 {
-    HttpResponseMessage httpResponseMessage;
-    std::wstring httpResponseBody;
-    try {
-        // Construct the HttpClient and Uri. This endpoint is for test purposes only.
-        HttpClient httpClient;
-        Uri requestUri{ L"http://localhost:7071/api/PostPushNotification" };
-
-        // Construct the JSON to post.
-        HttpStringContent jsonContent(
-            BuildNotificationPayload(channel.c_str()),
-            UnicodeEncoding::Utf8,
-            L"application/json");
-
-        // Post the JSON, and wait for a response.
-        httpResponseMessage = httpClient.PostAsync(
-            requestUri,
-            jsonContent).get();
-
-        // Make sure the post succeeded, and write out the response.
-        httpResponseMessage.EnsureSuccessStatusCode();
-        httpResponseBody = httpResponseMessage.Content().ReadAsStringAsync().get();
-        std::wcout << httpResponseBody.c_str() << std::endl;
+    // winrt::guid remoteId { L"a2e4a323-b518-4799-9e80-0b37aeb0d225" };
```

Comment:
Move closer to `channelOperation`.

Reply:
Introduced a global remoteId, we don't need these comments now